packages/git-diff/lib/git-diff-view.js

```diff
@@ -72,122 +151,106 @@ module.exports = class GitDiffView {
     this.moveToLineNumber(nextDiffLineNumber);
   }
 
-  updateIconDecoration() {
-    const gutter = this.editor.getElement().querySelector('.gutter');
-    if (gutter) {
-      if (
-        atom.config.get('editor.showLineNumbers') &&
-        atom.config.get('git-diff.showIconsInEditorGutter')
-      ) {
-        gutter.classList.add('git-diff-icon');
-      } else {
-        gutter.classList.remove('git-diff-icon');
-      }
-    }
-  }
-
   moveToPreviousDiff() {
     const cursorLineNumber = this.editor.getCursorBufferPosition().row + 1;
-    let previousDiffLineNumber = -1;
-    let lastDiffLineNumber = -1;
-    if (this.diffs) {
-      for (const { newStart } of this.diffs) {
-        if (newStart < cursorLineNumber) {
-          previousDiffLineNumber = Math.max(
-            newStart - 1,
-            previousDiffLineNumber
-          );
-        }
-        lastDiffLineNumber = Math.max(newStart - 1, lastDiffLineNumber);
+    let previousDiffLineNumber = null;
+    let lastDiffLineNumber = null;
+    for (const { newStart } of this.diffs) {
+      if (newStart < cursorLineNumber) {
+        previousDiffLineNumber = Math.max(newStart - 1, previousDiffLineNumber);
       }
+      lastDiffLineNumber = Math.max(newStart - 1, lastDiffLineNumber);
     }
 
     // Wrap around to the last diff in the file
     if (
       atom.config.get('git-diff.wrapAroundOnMoveToDiff') &&
-      previousDiffLineNumber === -1
+      previousDiffLineNumber === null
     ) {
       previousDiffLineNumber = lastDiffLineNumber;
     }
 
     this.moveToLineNumber(previousDiffLineNumber);
   }
 
-  moveToLineNumber(lineNumber) {
-    if (lineNumber != null && lineNumber >= 0) {
-      this.editor.setCursorBufferPosition([lineNumber, 0]);
-      this.editor.moveToFirstCharacterOfLine();
+  updateIconDecoration() {
+    const editorElement = atom.views.getView(this.editor);
+    const gutter = editorElement.querySelector('.gutter');
+    if (gutter) {
+      if (
+        atom.config.get('editor.showLineNumbers') &&
+        atom.config.get('git-diff.showIconsInEditorGutter')
+      ) {
+        gutter.classList.add('git-diff-icon');
+      } else {
+        gutter.classList.remove('git-diff-icon');
+      }
     }
   }
 
-  subscribeToRepository() {
-    this.repository = repositoryForPath(this.editor.getPath());
-    if (this.repository) {
-      this.subscriptions.add(
-        this.repository.onDidChangeStatuses(() => {
-          this.scheduleUpdate();
-        })
-      );
-      this.subscriptions.add(
-        this.repository.onDidChangeStatus(changedPath => {
-          if (changedPath === this.editor.getPath()) this.scheduleUpdate();
-        })
-      );
+  moveToLineNumber(lineNumber) {
+    if (lineNumber != null) {
```

Comment:
Are we sure that lineNumber is always more than 0?

Reply:
A buffer only has 0-N array slots. It wouldn't make sense to use negative integer points to represent a buffer position unless it was relative to something else inside the same buffer. It's not in the [Atom Docs](https://flight-manual.atom.io/api/v1.54.0/TextEditor/#instance-getCursorBufferPosition). You're a maintainer right? You tell me. If it can be negative, add it to the docs please, so people like me can know to guard for it.

Reply:
Yes, I modified the algorithms in the `moveToNextDiff` and `moveToPreviousDiff` methods to both return `null` or a positive integer. At least within Atom's spec. It's to my understanding that `Atom.TextEditor.getCursorBufferPosition()` returns a `Point`. It wouldn't make sense to return a negative graph `Point` using said method because buffer memory accesses are intuitively linear. Admittedly this is a bit presumptuous of me, and we all know deterministic behavior is better. So I'll look into the Atom internals in a bit to figure out if this is actually an issue and if it is, I'll update the docs so nobody else has to apply intuition for that method. I'll also update this thread with my findings either way so we're all on the same page.

But right now I have a COVID test to attend to.

Reply:
So `Atom.TextEditor.getCursorBufferPosition` eventually calls `Atom.TextBuffer.getMarkerStartPosition` containing the code:
```
Point.fromObject(@index.getStart(id))
```
*@index* is a `superstring` library method `marker` which only fetches a positive `row` and `column`.